README.md

```diff
@@ -1,94 +1,98 @@
-
-The user (once logged in JupyterHub) can use the `ProxyTokenHandler` (under the
-url `http(s)://JUPYTERHUBHOST/hub/proxytoken/`) to get the current token (via GET requests) or to ask to change it (via POST requests).
-This endpoint requires to be authenticated.
-
-To see an example, load inside your jupyter the notebook provided under `examples/Proxy token manager.ipynb`.
-Run it and then press the buttons to get the current token, or change it (disabling all access, enabling it for everybody, or generating a new random token).
-
-**NOTE**: this needs to be done inside the jupyter provided by JupyterHub,
-otherwise the JupyterHub authorization cookies will not be passed and you
-will not be able to access these endpoints.
-
-To use it:
-- Login in JupyterHub
-- (Optional, if you did not choose the option in the `jupyterhub_config.py` to generate a random token at startup) Upload the `examples/Proxy token manager.ipynb`, run it and create a new token
-- Open a terminal and start a server serving on one of the ports you chose (e.g.:  `python -m SimpleHTTPServer 5000`)
-- Try to connect to it. If you set it without token check (allow all), you can 
-  just go to the correct URL in your browser (e.g. `http://localhost:8000/hub/proxy5000/YOURJUPYTERHUBUSERNAME`). Otherwise, to check using a proxy
-  via a AJAX request (that checks that also the CORS headers are properly
-  set in the server), use the simple example under `examples/client_ajax_CORS_example.html`: put the correct URL (change the string `USERNAMEHERE`, and possibly change the token, that you can get as
-  described above via the Jupyter notebook, or via the link provided in the 
-  page, if you have already logged into JupyterHub in the same browser and you have not changed the URL of the `/proxytoken/` endpoint).
-
-### Notes
-- If the Docker host is a Mac, you need to start the server to be proxied 
-  inside docker on the 0.0.0.0 interface, otherwise Docker will not allow to 
-  forward it due to the way networking is configured by default on Docker 
-  on the Mac. Note that this is a Docker configuration and not a jhproxy issue.
+A safe way to proxy ports from docker containers running inside a JupyterHub to the outside world.
+
+## Features
+
+The JupyterHub administrator can:
+
+ * select which ports can be forwarded
+ * select whether to protect access to forwarded ports by tokens or not  
+   (when access protected, tokens are provided by `X-Proxy-Token` HTTP header)
+
+A user logged in to the JupyterHub can:
+
+ * get his or her current proxy token via a `GET` request to `http(s)://<JUPYTERHUBHOST>/hub/proxytoken/`
+ * disable or regenerate his or her proxy token via a `POST` request to `http(s)://<JUPYTERHUBHOST>/hub/proxytoken/`
+
+An internet user can:
+
+* access the proxied service at `http(s)://<JUPYTERHUBHOST>/hub/proxy/<PORT>/<USER>`
+
+
+
+## Limitations
+
+ * works (only) with `DockerSpawner` so far (extended to `jhproxy.spawners.TokenizedDockerSpawner`)
```

Comment:
This is optional, if you use simply DockerSpawner it should (untested) just work, but it's always in 'allow all' mode (it's anyway already written below, so maybe ok to leave as is